test/e2e/webhook_test.go

```diff
@@ -427,6 +429,10 @@ var _ = Describe("Fleet's Hub cluster webhook tests", func() {
 
 				By(fmt.Sprintf("expecting admission of operation CREATE of ReplicaSet in reserved namespace %s", ns.Name))
 				Expect(HubCluster.KubeClient.Create(ctx, rs)).Should(Succeed())
+
+				By("clean up created replica set")
+				Expect(HubCluster.KubeClient.Get(ctx, types.NamespacedName{Name: rsName, Namespace: ns.Name}, rs)).Should(Succeed())
+				Expect(HubCluster.KubeClient.Delete(ctx, rs))
```

Comment:
Hi Arvind! For the delete part, to avoid conflicts, it would be nice to have an eventually block to make sure that the object does disappear.